src/ucp/core/ucp_ep.c

```diff
@@ -394,7 +395,13 @@ ucs_status_t ucp_ep_create_base(ucp_worker_h worker, unsigned ep_init_flags,
         ucp_ep_update_flags(ep, UCP_EP_FLAG_INTERNAL, 0);
         ucs_list_add_tail(&worker->internal_eps, &ep->ext->ep_list);
     } else {
-        ucs_list_add_tail(&worker->all_eps, &ep->ext->ep_list);
+        if (worker->context->config.ext.shuffle_worker_eps) {
+            new_ep_index = ucs_rand() % (worker->num_all_eps + 1);
+            ucs_list_insert_at_index(&worker->all_eps, &ep->ext->ep_list,
+                                    new_ep_index);
```

Comment:
this will cause O(n^2) initialization time
better solution approach could be to shuffle the list using fisher yates method every N flush operations, and only if new EPs were added since the previous shuffle. .

Reply:
Got a simpler one. Insert each time at head/tail with 50% probability. Will lead to random result, with zero cost.
wdyt?

Reply:
Regarding the coin-flip method for insertion, I looked at the standard deviation of the positions, and it increases as the number of the element gets higher. 
Numbers that are inserted early (like 1 and 2) tend to cluster more around the middle of the list, while numbers that are inserted later (like 9 and 10) are more widely dispersed, resulting in a higher standard deviation. 
The distribution of the resulting list isn’t completely uniform, so I guess it depends on how much randomness the customer requires.
Please let me know how to proceed.
@gleon99 @yosefe 
![image](https://github.com/user-attachments/assets/cfa5ba2e-4124-4001-8f1a-d755910728ff)
![image](https://github.com/user-attachments/assets/06563d15-04ce-4e8d-b5c1-a52ca6815d98)